### PR TITLE
Replace GitHubData lib code with OctoKit

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

💡For more information on branching, refer to the project's [branching documentation](../../Documentation/Branching.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the title of the associated issue.

- [x] A `preview` label has been added to the pull request.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

File: VersionMiner/App.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+namespace VersionMiner;
+
+/// <summary>
+/// Provides simple application type information.
```
Comment (KinsonDigitalAdmin) [inline VersionMiner/App.cs:11]: Refactor to below:

```suggestion
/// Provides simple application information.
```

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — DISMISSED** [review]

File: VersionMiner/Services/IHttpClient.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    /// </summary>
+    ReadOnlyCollection<(string name, string value)> DefaultHeaders { get; }
+
+    /// <summary>
+    /// Adds a header to the next request using the given <paramref name="name"/> and <paramref name="value"/>.
+    /// </summary>
+    /// <param name="name">The name of the header to add.</param>
+    /// <param name="value">The value of the header to add.</param>
+    /// <remarks>
+    ///     After the next request is made, the headers will be cleared out.
```
Comment (kselena) [inline VersionMiner/Services/IHttpClient.cs:35]: Grammar - can't end sentence w/ a preposition
```suggestion
    ///     After the next request is made, the headers will be cleared.
```

File: VersionMiner/Exceptions/HeaderAlreadyExistsException.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+// <copyright file="HeaderAlreadyExistsException.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+namespace VersionMiner.Exceptions;
+
+/// <summary>
+/// Occurs when an HTTP header already exists.
```
Comment (kselena) [inline VersionMiner/Exceptions/HeaderAlreadyExistsException.cs:8]: Grammar
```suggestion
/// Occurs when an HTTP header exists.
```

File: VersionMiner/Services/HttpClient.cs [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+    {
+        foreach ((string name, string value) header in headers)
+        {
+            AddRequestHeader(header.name, header.value);
+        }
+    }
+
+    /// <inheritdoc/>
+    /// <exception cref="Exception">
+    ///     Invoked when a default header already exists.
```
Comment (kselena) [inline VersionMiner/Services/HttpClient.cs:101]: Grammar
```suggestion
    ///     Invoked when a default header exists.
```

File: VersionMiner/Services/HttpClient.cs [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+    /// </exception>
+    public void AddDefaultHeader(string name, string value) => this.restClient.AddDefaultHeader(name, value);
+
+    /// <inheritdoc/>
+    /// <exception cref="Exception">
+    ///     Invoked when a default header already exists.
```
Comment (kselena) [inline VersionMiner/Services/HttpClient.cs:107]: Grammar
```suggestion
    ///     Invoked when a default header exists.
```

File: VersionMiner/Services/HttpClient.cs [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+        {
+            return;
+        }
+
+        this.nextRequestHeaders.Add((name, value));
+    }
+
+    /// <inheritdoc/>
+    /// <exception cref="Exception">
+    ///     Invoked when a default header already exists.
```
Comment (kselena) [inline VersionMiner/Services/HttpClient.cs:89]: Grammar
```suggestion
    ///     Invoked when a default header exists.
```

File: VersionMiner/Services/HttpClient.cs [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+                result.Add((parameter.Name, (string)parameter.Value));
+            }
+
+            return new ReadOnlyCollection<(string name, string value)>(result.ToArray());
+        }
+    }
+
+    /// <inheritdoc/>
+    /// <exception cref="Exception">
+    ///     Invoked when a default header already exists.
```
Comment (kselena) [inline VersionMiner/Services/HttpClient.cs:75]: Grammar
```suggestion
    ///     Invoked when a default header exists.
```

File: VersionMiner/Services/IHttpClient.cs [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    /// Removes the first occurence of a header that matches the given header <paramref name="name"/>.
+    /// </summary>
+    /// <param name="name">The name of the header.</param>
+    /// <remarks>
+    ///     Multiple headers with the same name are allowed in an HTTP request.
+    /// </remarks>
+    void RemoveHeader(string name);
+
+    /// <summary>
+    /// Clears all of the currently existing headers.
```
Comment (kselena) [inline VersionMiner/Services/IHttpClient.cs:73]: Grammar
```suggestion
    /// Clears all of the existing headers.
```

**Review by kselena — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Review by kselena — APPROVED** [review]
